File: factoryDP/AnimalPackage/AnimalFactory.java
package AnimalPackage;

public class AnimalFactory{
    public Animal makeAnimal(String animalType){
        Animal animal = null;

        if(animalType.equals("Dog")){
            animal = new Dog();
        }
        else if(animalType.equals("Cat")){
            animal = new Cat();
        }
        else if(animalType.equals("Lion")){
            animal = new Lion();
        }
        else if(animalType.equals("Horse")){
            animal = new Horse();
        }
        else if(animalType.equals("Cow")){
            animal = new Cow();
        }
        return animal;
    }
}

File: factoryDP/cpp/AnimalFactory.cpp
#include "AnimalFactory.hpp"

Animal* AnimalFactory :: generateAnimal(string animalType){
    Animal *animal;
    if(animalType == "Cat"){
        animal = new Cat();
    }
    else if(animalType == "Dog"){
        animal = new Dog();
    }
    else if(animalType == "Horse"){
        animal = new Horse();
    }
    return animal;
}